Tools/autotest/arducopter.py

```diff
@@ -14317,8 +14330,15 @@ def LuaParamSet(self):
         self.reboot_sitl()
         self.assert_parameter_value('DISARM_DELAY', 111)
 
-        # very bad things happen if we don't turn things off at the end..
-        self.set_parameter("PARAM_SET_ENABLE", 0)
+        # set parameters to their original value so the following
+        # context pop should only need to set PARAM_SET_ENABLE back to
+        # its default value of 1.  That prevents permission errors
+        # created by param-set.lua!
+        self.set_parameters(orig_parameter_values)
+
+        self.context_pop()
+
+        self.reboot_sitl()
```

Comment:
```suggestion
```

Above context pop reboots since we rebooted after the corresponding push.

Reply:
Good point, removed this.  The context-magically-reboots stuff is relatively new.